test/integration/acceptance/add_peer_test.cpp

```diff
@@ -0,0 +1,325 @@
+ public:
+  using FakePeer = fake_peer::FakePeer;
+
+  std::unique_ptr<IntegrationTestFramework> itf_;
+
+  /**
+   * Create honest fake iroha peers
+   *
+   * @param num_fake_peers - the amount of fake peers to create
+   */
+  void createFakePeers(size_t num_fake_peers) {
+    fake_peers_ = itf_->addFakePeers(num_fake_peers);
+  }
+
+  /**
+   * Prepare state of ledger:
+   * - create account of target user
+   * - add assets to admin
+   *
+   * @return reference to ITF
+   */
+  IntegrationTestFramework &prepareState() {
+    itf_->setGenesisBlock(itf_->defaultBlock()).subscribeQueuesAndRun();
+
+    auto permissions =
+        interface::RolePermissionSet({Role::kReceive, Role::kTransfer});
+
+    return itf_
+        ->sendTxAwait(makeUserWithPerms(permissions), checkBlockHasNTxs<1>)
+        .sendTxAwait(
+            complete(baseTx(kAdminId).addAssetQuantity(kAssetId, "20000.0"),
+                     kAdminKeypair),
+            checkBlockHasNTxs<1>);
+  }
+
+ protected:
+  void SetUp() override {
+    itf_ =
+        std::make_unique<IntegrationTestFramework>(1, boost::none, true, true);
+    itf_->initPipeline(kAdminKeypair);
+  }
+
+  std::vector<std::shared_ptr<FakePeer>> fake_peers_;
+};
+
+auto makePeerPointeeMatcher(interface::types::AddressType address,
+                            interface::types::PubkeyType pubkey) {
+  return ::testing::Truly(
+      [address = std::move(address),
+       pubkey = std::move(pubkey)](std::shared_ptr<interface::Peer> peer) {
+        return peer->address() == address and peer->pubkey() == pubkey;
+      });
+}
+
+auto makePeerPointeeMatcher(std::shared_ptr<interface::Peer> peer) {
+  return makePeerPointeeMatcher(peer->address(), peer->pubkey());
+}
+
+/**
+ * @given a network of single peer
+ * @when it receives a valid signed addPeer command
+ * @then the transaction is committed
+ *    @and the WSV reports that there are two peers: the initial and the added one
+ */
+TEST_F(FakePeerExampleFixture, FakePeerIsAdded) {
+  // init the real peer with no other peers in the genesis block
+  auto &itf = prepareState();
+
+  const std::string new_peer_address = "127.0.0.1:1234";
+  const auto new_peer_pubkey =
+      shared_model::crypto::DefaultCryptoAlgorithmType::generateKeypair()
+          .publicKey();
+
+  itf.sendTxAwait(
+      complete(baseTx(kAdminId).addPeer(new_peer_address, new_peer_pubkey),
+               kAdminKeypair),
+      checkBlockHasNTxs<1>);
+
+  auto opt_peers = itf.getIrohaInstance()
+                       .getIrohaInstance()
+                       ->getStorage()
+                       ->createPeerQuery()
+                       .value()
+                       ->getLedgerPeers();
+
+  ASSERT_TRUE(opt_peers);
+  EXPECT_THAT(*opt_peers,
+              ::testing::UnorderedElementsAre(
+                  makePeerPointeeMatcher(itf.getThisPeer()),
+                  makePeerPointeeMatcher(new_peer_address, new_peer_pubkey)));
+}
```

Comment:
I think there could be also a test or an extension of this one which checks that `ledger_state` from synchronizer contains new peer.